linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ModuleLinkerOutput.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+/** A single Module output specification for a linker run.
+ *
+ *  The fields [[moduleID]] and [[isInternal]] uniquely identify this output in
+ *  a [[Linker]] run.
+ *
+ *  @param moduleID The ID of the module this linker output represents.
+ *
+ *  @param isInternal Whether this is an internal (or an external) module.
+ *
+ *  @param output The [[LinkerOutput]] for this module.
+ *
+ *  @param moduleName Name of this module for import. A [[Linker]] will use
+ *      this name to include this module from one of its dependees. Must be set,
+ *      if this module is internal.
```

Comment:
I do not really understand what `moduleName` does. Is this basically the relative path of this module as seen from the modules that will depend on it? I guess that assumes that all modules are in the same directory?

What is the meaning of `moduleName` for an external module?

Reply:
`moduleName` is the `"module-name"` of the import statement ([MDN][1]). It is used in the modules depending on this one to import it. This is why it is required for internal modules but not external modules. For external modules, this would be ignored but not harmful (thought is that it might simplify upstream code if we allow setting it all the time).

This is similar to the `URI`s in `LinkerOutput`: We allow to set them partially even if it is "typically meaningless".

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import

Reply:
OK, I see.